cleanrl/ppo_pettingzoo_ma_atari.py

```diff
@@ -219,6 +227,8 @@ def get_action_and_value(self, x, action=None):
             next_value = agent.get_value(next_obs).reshape(1, -1)
             advantages = torch.zeros_like(rewards).to(device)
             lastgaelam = 0
+            next_done = np.logical_or(next_termination, next_truncation)
```

Comment:
I think there is bug here. We should still bootstrap if `next_truncation=True` :

> "you should bootstrap if infos[env_idx]["TimeLimit.truncated"] is True (episode over due to a timeout/truncation) or dones[env_idx] is False (episode not finished)." - [stable baselines](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html)

So `next_done=next_termination` and  `dones=terminations` (probs just use next_terminations and terminations directly). 

To implement this correctly we also need access to `terminal_observation` from `pettingzoo_env_to_vec_env_v1`. I have a [PR](https://github.com/Farama-Foundation/SuperSuit/pull/233) out for this. Then we can implement something like [this](https://github.com/DLR-RM/stable-baselines3/blob/69afefc91d408d352b4224ae5244ad2c32bb7634/stable_baselines3/common/on_policy_algorithm.py#L212) to do correct bootstrapping for truncating/timeout.

Reply:
Good catch @KaleabTessera would you be willing to update this branch with the changes? I can give you edit access, I currently have a lot of other obligations from work so don’t have much time for this

Reply:
Oh shoot it’s a patch-1 so I don’t know if you can be given access. But if you clone the repo you can make a new branch from this branch and make a new PR if it’s not possible to edit this branch? Or maybe make a PR to update this branch itself. Sorry I can’t help more

Reply:
FYI I am doing a refactor at #424 . Gonna try run a whole suite of benchmark soon.

Reply:
> I think there is bug here. We should still bootstrap if `next_truncation=True` :
> 
> > "you should bootstrap if infos[env_idx]["TimeLimit.truncated"] is True (episode over due to a timeout/truncation) or dones[env_idx] is False (episode not finished)." - [stable baselines](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html)
> 
> So `next_done=next_termination` and `dones=terminations` (probs just use next_terminations and terminations directly e.g. `nextnonterminal = 1.0 - next_termination` ).
> 
> To implement this correctly we also need access to `terminal_observation` from `pettingzoo_env_to_vec_env_v1` since we need access to the true terminal obs and not the obs returned by the next restart (the case currently -- so we need infos to provide access to the terminal obs). I have a [PR](https://github.com/Farama-Foundation/SuperSuit/pull/233) out for this . Then we can implement something like [this](https://github.com/DLR-RM/stable-baselines3/blob/69afefc91d408d352b4224ae5244ad2c32bb7634/stable_baselines3/common/on_policy_algorithm.py#L212) to do correct bootstrapping for truncating/timeout.

Btw, just as an update, the SuperSuit PR linked has been merged. My only concern with this is that whatever bootstrapping behavior is done here should mirror what is done with the single agent PPO implementations, so this is a question for @vwxyzjn. 

My inclination is to keep the logic as it currently is in this PR and address that bootstrapping issue in another PR (maybe @KaleabTessera is interested in doing that? I don't have a whole lot of time to look into it nor am I the best person to do it as I'm not an expert)